projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.spec.ts

```diff
@@ -0,0 +1,195 @@
+import { OrderApprovalDetailFormComponent } from './order-approval-detail-form.component';
+
+const REJECT = 'REJECT';
+const APPROVE = 'APPROVE';
+
+const mockOrderApproval = {
+  approvalDecisionRequired: true,
+  code: '00000005',
+  order: {
+    code: '00000001',
+    totalPriceWithTax: {
+      formattedValue: '$9,609.99',
+    },
+  },
+} as OrderApproval;
+
+class MockOrderApprovalDetailService {
+  getOrderApproval(): Observable<OrderApproval> {
+    return of(mockOrderApproval);
+  }
+}
+
+@Component({
+  selector: 'cx-form-errors',
+  template: '',
+})
+class MockFormErrorsComponent {
+  @Input()
+  controll: FormControl;
+}
+
+@Pipe({
+  name: 'cxUrl',
+})
+class MockUrlPipe implements PipeTransform {
+  transform() {}
+}
+
+class MockOrderApprovalService {
+  makeDecision() {}
+}
+
+describe('OrderApprovalDetailFormComponent', () => {
+  let component: OrderApprovalDetailFormComponent;
+  let fixture: ComponentFixture<OrderApprovalDetailFormComponent>;
+  let orderApprovalDetailService: OrderApprovalDetailService;
+  let orderApprovalService: OrderApprovalService;
+  let el: DebugElement;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        OrderApprovalDetailFormComponent,
+        MockFormErrorsComponent,
+        MockUrlPipe,
+      ],
+      imports: [ReactiveFormsModule, I18nTestingModule],
+      providers: [
+        {
+          provide: OrderApprovalDetailService,
+          useClass: MockOrderApprovalDetailService,
+        },
+        { provide: OrderApprovalService, useClass: MockOrderApprovalService },
+      ],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(OrderApprovalDetailFormComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+    orderApprovalDetailService = TestBed.inject(OrderApprovalDetailService);
+    orderApprovalService = TestBed.inject(OrderApprovalService);
+    el = fixture.debugElement;
+  });
+
+  it('should create', () => {
+    console.log('temp', orderApprovalDetailService, orderApprovalService);
+    expect(component).toBeTruthy();
+  });
+
+  it('should display approval form when approve order button clicked, and then hide form on cancel.', () => {
+    displayAndCancelDecisionForm(APPROVE);
+  });
+
+  it('should display rejection form when reject order button clicked, and then hide form on cancel.', () => {
+    displayAndCancelDecisionForm(REJECT);
+  });
+
+  it('should submit approval form.', () => {
+    displayDecisionForm(APPROVE);
+    submitDecisionForm(APPROVE);
+  });
+
+  it('should submit rejection form.', () => {
+    displayDecisionForm(REJECT);
+    submitDecisionForm(REJECT);
+  });
+
+  it('should comment be optional for approval.', () => {
+    displayDecisionForm(APPROVE);
+    expect(component.approvalForm.valid).toBeTrue();
+  });
+
+  it('should comment be required for rejection.', () => {
```

Comment:
same as above, but maybe we can add an added assertion where we click it and it validation is still false